### DISCUSSION TITLE
Execution Provider: `Ort` prefix of not?

### DISCUSSION THREAD

**Discussion by fgervais** [microsoft/onnxruntime#12641] [Other Q&A]
I don't quite get execution providers.

In the [documentation](https://onnxruntime.ai/docs/execution-providers/add-execution-provider.html#use-the-execution-provider), I see all functions have an `Ort` prefix.

However for example in the [triton onnxruntime backend](https://github.com/triton-inference-server/onnxruntime_backend/blob/4cddb165a302a368fb3e5ccd277816a4b0c9fbc9/src/onnxruntime.cc#L582), they use functions without the prefix.

Then if I search through the onnxruntime source code, I see some functions for example `SessionOptionsAppendExecutionProvider_OpenVINO` have code with and without the prefix but some other like `SessionOptionsAppendExecutionProvider_TensorRT_V2` doesn't have definitions with the `Ort` prefix.

Could I get some help to make sense of this?

**Comment by jywu-mysoft** [comment]
Apologies. That document is pretty outdated. We will update it.

There is no specific requirement/convention that the api functions are prefixed with Ort. Many of them started with that prefix, but as the api evolved and new functions were added, many of the newer ones don't have the Ort prefix (as it's kind of redundant. we keep the old api's to maintain compatibility with older ORT version and introduce new ones without the Ort prefix).
You may see different versions of functions for "AppendExecutionProvider"
For example, you'll see in the header that there is both a SessionOptionsAppendExecutionProvider_TensorRT and SessionOptionsAppendExecutionProvider_TensorRT_V2. The latter is the newer version (and used by Triton) The reason we added a new one is the older SessionOptionsAppendExecutionProvider_TensorRT  takes a publicly defined struct OrtTensorRTProviderOptions. Every time we add a new field to that struct, we technically break ABI compatibility. (so we really should create a new version of the struct. but that would require a new interface every time we created a new version)
SessionOptionsAppendExecutionProvider_TensorRT_V2 was created to take in an opaque struct OrtTensorRTProviderOptionsV2 which the user does not directly access. see more details in the comments at https://github.com/microsoft/onnxruntime/blob/52ce6a90b4b52e0b331e6c0c61e05478589bf108/include/onnxruntime/core/session/onnxruntime_c_api.h#L2743
Other Execution Provider api's for AppendExecutionProvider follow this new pattern but not all of the Execution Provider api's have been updated (as they may have different internal/external maintainers)
